firmware/arduino/chessboard/motorDriver.ino

```diff
@@ -123,6 +147,13 @@ uint8_t moveStraight(int motor[], int startCol, int startRow, int endCol, int en
 
   numSteps = unitSpaces * stepsPerUnitSpace;
 
+  // Setting numEighthSteps to 8 because function only moves in whole steps
+  // Sign is set based off of direction of each motor
+  numEighthStepsX = 8;
+  numEighthStepsY = 8;
+  numEighthStepsX *= (dir == RIGHT) ? 1 : -1;
+  numEighthStepsY *= (dir == UP) ? 1 : -1;
```

Comment:
The way this is currently, you're using the same value of dir for the X and Y motion and updating both x and y position regardless of which direction you're moving.

`numEighthStepsX` and `numEighthStepsY` should be set inside the if statements. If x motion, do this:
```c
numEighthStepsX = (dir == RIGHT) ? 8 : -8;
numEighthStepsY = 0;
```
if y motion do this:
```c
numEighthStepsX = 0;
numEighthStepsY = (dir == UP) ? 8 : -8;
```

That way, you don't need to change the code below where you have
```c
// Updating current position per step
currentX += numEighthStepsX;
currentY += numEighthStepsY;
```
since if you're doing x motion, you're adding 0 to y and vice versa.

Alternatively, you can do something similar to what you did in `homeAxis` and use a pointer to update only one of the current positions, and you would set the value of that pointer in your if statements. Do you have a preference @LSaber36 ?

Reply:
For consistency, we should mimic what we have in homeAxis() and use a pointer to refer to either of the directions. I didn't even realize the redundancy there, thanks!

Reply:
I agree with Lucas, let us know if you need help getting that worked out